app.py

```diff
@@ -0,0 +1,125 @@
+import logging
+import sys
+import threading
+import time
+import typing
+from dataclasses import dataclass
+
+from fastapi import FastAPI
+from fastapi.middleware.cors import CORSMiddleware
+import json
+import requests
+import uvicorn
+import yaml
+
+from args import get_args
+
+
+app = FastAPI()
+args = get_args()
+
+with open(args.config, "r") as stream:
+    try:
+        data = yaml.safe_load(stream)
+        API_KEY = data.get("api_key", None)
+        stops = data.get("stops", {})
+    except Exception:
+        logging.exception("unable to open yaml file/ file is missing data, exiting")
+        sys.exit(1)
+
+logging.Formatter.converter = time.gmtime
+
+formatter = logging.basicConfig(
+    format="%(asctime)s.%(msecs)03dZ %(threadName)s %(levelname)s:%(name)s:%(message)s",
+    datefmt="%Y-%m-%dT%H:%M:%S",
+    level=logging.INFO,
+)
+
+origins = ["*"]
+app.add_middleware(
+    CORSMiddleware,
+    allow_origins=origins,
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+@dataclass
+class Prediction:
+  route: str
+  times: typing.List[str]
+
+@dataclass
+class Stop:
+  id: str
+  name: str
+  predictions: typing.List[Prediction]
+
+PREDICTIONS_URL = 'https://api.511.org/transit/StopMonitoring'
+cache = []
+def update_cache():
+    # clear cache of any previous (now outdated) data
+    cache.clear()
+    # send post request for each agency's stop(s)
+    for stop in stops:
+        agency = stop.get('operator')
+        stopCode = stop.get('id')
+        stopName= stop.get('name')
+
+        params = {
+            'api_key': API_KEY,
+            'agency': agency,
+            'stopCode': stopCode,
+            'format': 'json'
+        }
+
+        response = requests.get(PREDICTIONS_URL, params=params)
+
+        if response.status_code != 200:
+            logging.error(f"not parsing response because response code was {response.status_code}")
+
```

Comment:
we log that we arent parsing but after the if statement, the code continues?

Reply:
fixed